tfjs-backend-webgl/src/kernels/Conv2D.ts

```diff
@@ -39,7 +39,8 @@ export function conv2d(
   if (convInfo.filterHeight === 1 && convInfo.filterWidth === 1 &&
       convInfo.dilationHeight === 1 && convInfo.dilationWidth === 1 &&
       convInfo.strideHeight === 1 && convInfo.strideWidth === 1 &&
-      (convInfo.padInfo.type === 'SAME' || convInfo.padInfo.type === 'VALID')) {
+      (convInfo.padInfo.type === 'SAME' || convInfo.padInfo.type === 'VALID') &&
+      $dataFormat === 'channelsLast') {
```

Comment:
For 1x1 conv2d branch,
if the data format is NHWC, it becomes `xReshaped` * `filterReshaped` with `transposeA` = `false` and `transposeB` = `false`.
If the data format is NCHW, it becomes
`filterReshaped` * `xReshaped` with `transposeA` = `true` and `transposeB` = `false`.
So you still can go to the `conv2dByMatMul` path with some changes.

Reply:
Thank you for your idea about the 1x1 conv2d branch! Just added some reshaping and transposing logics to support the `NCHW` format. Also, added some basic tests for `NCHW` format and a test for `NCHW` format with batch=2.

We already have a [test](https://github.com/tensorflow/tfjs/blob/ff9910e898735e0a7ebe8f5e214bf3e872d4b7aa/tfjs-core/src/ops/conv2d_test.ts#L340-L359) of `NCHW` format for the third path.